pkg/ring/kv/client.go

```diff
@@ -149,10 +151,10 @@ func createClient(name string, prefix string, cfg StoreConfig, codec codec.Codec
 		client = PrefixClient(client, prefix)
 	}
 
-	return metrics{client}, nil
+	return newMetricsClient(name, backend, client, reg), nil
```

Comment:
If `reg` is nil (which is a case we should support and is typical for tests), we can skip creation of metrics client completely.

Reply:
👍 Make sense

Reply:
Makes sense